api/server/router/network/filter.go

```diff
@@ -29,6 +29,29 @@ func filterNetworkByType(nws []types.NetworkResource, netType string) ([]types.N
 	return retNws, nil
 }
 
+func filterNetworkByUse(nws []types.NetworkResource, danglingOnly bool) ([]types.NetworkResource, error) {
+	retNws := []types.NetworkResource{}
+
+	switch danglingOnly {
```

Comment:
**NOTE**: This is a suggestion.
I would prefer get rid of the `switch` block and do the following:
```go
for _, nw := range nws {
    if !runconfig.IsPreDefinedNetwork(nw.Name) && len(nw.Containers) == 0 && danglingOnly {
        retNws = append(retNws, nw)
    }
}
```
I think it'll work.